### ISSUE TITLE
Cloud saving / sync

### ISSUE THREAD

**Issue by borekb** [mike-ward/Markdown-Edit#58]
Feature suggestion: let me connect my Google Drive / DropBox to MDE and pick a remote folder that will be used as a save / sync location for Markdown documents. StackEdit does this rather nicely.

One additional challenge for MDE is how to treat documents that are both local and stored in the cloud (StackEdit doesn't have this issue as it doesn't have access to local file system). I think MDE should somehow store the connection between a local path and remote document name and if I open the local file, start storing into that remote document, and if I "open from cloud" (new menu item), start saving (Ctrl+S) into this local path. But this might be a little bit tricky.

**Comment by mike-ward** [comment]
Doesn't Dropbox and Google Drive already do file syncing? If so, would it just be a matter of opening the file from the sync'ed folder? I don't use these services so I'm not entirely sure how they work.

**Comment by borekb** [comment]
Yes that would work IF my project / Markdown files were inside the synced folder. I cannot afford that, however - OneDrive (in my case) requires everything to be under c:\users\xyz\OneDrive and my dev projects are not there (it wouldn't be practical - their sync icon overlays would conflict with TortoiseGit overlays, it would constantly be uploading something as I compile my projects etc.)

If MDE could work with some cloud storage provider via their API rather than via locally installed application, that would great.

**Comment by mike-ward** [comment]
To do it right would require an almost `Git' type of syncing. Otherwise, you have to pick a winner when there's a conflict between the remote and local versions. And then there's the issue of different local copies on different machines. I'm not ruling this out just yet, but there are some difficult issues to resolve.

**Comment by borekb** [comment]
I agree. The simplest way to handle conflicts is to ask user which version to keep but sure, it's not the easiest feature to implement. Just answering the call for "big ideas" for next features from the 0.13 release notes :)

**Comment by mike-ward** [comment]
I'm not being critical of the idea. Just thinking out loud.

**Comment by justdmitry** [comment]
Editor should edit files, not sync them.

If you want 'hack' OneDrive and force him sync files/folders outside of default sync folder - you can use hardlink / junction points feature of NTFS filesystem (I use [Link Shell extensions](http://schinagl.priv.at/nt/hardlinkshellext/linkshellextension.html) for that) or use other cloud sync service that meet your needs ([Cubby](https://www.cubby.com) for example)
